aopy/data/bmi3d.py

```diff
@@ -2663,6 +2663,168 @@ def tabulate_behavior_data_random_targets(preproc_dir, subjects, ids, dates, met
+            | **periph_display_time (ntrial):** time at which the peripheral target was displayed
+            | **periph_target_displayed (ntrial):** boolean values indicating whether the peripheral target was displayed
+            | **auditory_start_time (ntrial):** time at which the set of tones starts 
+            | **go_cue_heard (ntrial):** boolean values indicating whether the go cue was heard
+            | **go_cue_time (ntrial):** time at which the go cue was presented
+            | **reach_completed (ntrial):** boolean values indicating whether the reach was completed
+            | **reach_end_time (ntrial):** time at which the reach was completed
+            | **reward_start_time (ntrial):** time at which the reward was presented
+            | **reward_end_time (ntrial):** time at which the reward was completed
+            | **penalty_start_time (ntrial):** time at which the penalty was presented
+            | **penalty_end_time (ntrial):** time at which the penalty was completed
+            | **penalty_event (ntrial):** numeric code for the penalty event
+    '''
+    # Use default "trial" definition
+    task_codes = load_bmi3d_task_codes()
+    trial_end_codes = [task_codes['TRIAL_END']]
+    trial_start_codes = [task_codes['CENTER_TARGET_ON']]
+    reward_codes = [task_codes['REWARD']]
+    penalty_codes = [task_codes['DELAY_PENALTY'], task_codes['HOLD_PENALTY'], 
+                     task_codes['TIMEOUT_PENALTY'], task_codes['OTHER_PENALTY']]
+    target_codes = task_codes['PERIPHERAL_TARGET_ON']
+    auditory_codes = task_codes['CUE']
+
+    always_meta = ['prepbuff_time' , 'delay_time'] #need these to calculate go cue timing 
+    # Concatenate base trial data
+    new_df = tabulate_behavior_data(
+        preproc_dir, subjects, ids, dates, trial_start_codes, trial_end_codes, 
+        reward_codes, penalty_codes, metadata = metadata+always_meta, df=None)
+    if len(new_df) == 0:
+        warnings.warn("No trials found")
+        return df
+
+    # Add target info
+    target_idx = [
+        code[np.isin(code, target_codes)][0] - target_codes[0] + 1 # add 1 for center target
+        if np.sum(np.isin(code, target_codes)) > 0 else 0 
+        for code 
+        in new_df['event_codes']
+    ]
+    target_location = [
+        np.squeeze(get_target_locations(preproc_dir, s, te, d, [t_idx]))
+        for s, te, d, t_idx 
+        in zip(new_df['subject'], new_df['te_id'], new_df['date'], target_idx)
+    ]
+    new_df['target_idx'] = target_idx
+    new_df['target_location'] = target_location
+
+    # Add trial segment timing
+    new_df['prev_trial_end_time'] = np.nan
+    new_df['trial_end_time'] = np.nan
+    new_df['center_target_on_time'] = np.nan
+    new_df['trial_initiated'] = False
+    new_df['hold_start_time'] = np.nan
+    new_df['hold_completed'] = False
+    new_df['periph_display_time'] = np.nan
+    new_df['periph_target_displayed'] = False
+    new_df['auditory_start_time'] = np.nan
```

Comment:
i think it might be helpful to split this out into ready_time, set_time, and go_time.
right now the set_time is missing so you have to know that it was 500 ms